src/Nethermind/Nethermind.Core.Test/Builders/BlockBuilder.cs

```diff
@@ -78,6 +78,11 @@ public BlockBuilder WithTransactions(int txCount, IReleaseSpec releaseSpec)
 
             return WithTransactions(txs);
         }
+
+        public BlockBuilder WithKnownTransactions(Transaction[] transactions)
+        {
+            return WithTransactions(transactions);
+        }
```

Comment:
What is the purpose there?

Reply:
I wanted to be able to build a block with transactions I pass in. The alternative looked like "WithTransactions" where I can create a certain number of transactions, but I don't get to specify gas prices and block numbers.

Maybe I should rename it to "WithGivenTransactions"?

Reply:
Actually I just realized that this is pointless haha, I could have just used WithTransactions to begin with